src/main/java/com/epam/reportportal/cucumber/Utils.java

```diff
@@ -253,13 +251,29 @@ public static TestCaseIdEntry getTestCaseId(Match match, String codeRef) {
 			return testCaseIdAnnotation != null ?
 					getTestCaseId(testCaseIdAnnotation, method, match.getArguments()) :
 					getTestCaseId(codeRef, match.getArguments());
-		} catch (NoSuchFieldException e) {
-			return getTestCaseId(codeRef, match.getArguments());
-		} catch (IllegalAccessException e) {
+		} catch (NoSuchFieldException | IllegalAccessException e) {
 			return getTestCaseId(codeRef, match.getArguments());
 		}
 	}
 
+	static List<ParameterResource> getParameters(Match match) {
+		List<ParameterResource> parameters = Lists.newArrayList();
+		Optional<String> parameterType = Optional.empty();
+
+		String text = match.getLocation();
+		Matcher matcher = Pattern.compile(PARAMETER_TYPE_REGEX).matcher(text);
+		if (matcher.find()) {
+			parameterType = Optional.of(text.substring(matcher.start() + 1, matcher.end() - 1));
```

Comment:
What about 2 parameters with different types, e.g. `Given I have <int> <string> in my pocket` (6, apples)?

Reply:
done